docs/tools.md

```diff
@@ -291,6 +291,43 @@ print(result.output)
 ```
 _(This example is complete, it can be run "as is")_
 
+For more complex scenarios where you need custom user prompts and tool return messages, you can use [`MultiModalToolResponse`][pydantic_ai.messages.MultiModalToolResponse]:
+
+```python {title="computer_use_example.py" test="skip" lint="skip"}
+import time
+from pydantic_ai import Agent
+from pydantic_ai.messages import MultiModalToolResponse, BinaryContent
+
+agent = Agent('openai:gpt-4o')
+
+@agent.tool_plain
+def click_and_compare(x: int, y: int) -> MultiModalToolResponse:
+    """Click at coordinates and show before/after screenshots."""
+    # Take screenshot before action
+    before_screenshot = capture_screen()
+
+    # Perform click operation
+    perform_click(x, y)
+    time.sleep(0.5)  # Wait for UI to update
+
+    # Take screenshot after action
+    after_screenshot = capture_screen()
+
+    return MultiModalToolResponse(
```

Comment:
While implemented with multi-modal content in mind, this can really be used to insert _any_ user content after the tool return. 

I'm also thinking a class like this could be used for https://github.com/pydantic/pydantic-ai/issues/2034, where the user wants to attach artifacts that won't be sent to the model but can be read from the `ToolReturnPart` programmatically.

So I'd like to name this more generically. What do you think of `ToolReturn`, to match `ToolReturnPart`?